core/include/xfa.h

```diff
@@ -158,14 +158,20 @@ _Pragma("GCC diagnostic ignored \"-Warray-bounds\"")
  *
  * Pointers will end up sorted by prio.
  *
+ * @note This implementation uses the __typeof__() C extension.
+ *       It is available both in GCC and LLCM, and both don't complain with
+ *       plain "-std=c11". If deemed necessary, use of __typeof__ can be removed
+ *       and the type can be added as parameter, at the cost of much less
+ *       convenience.
```

Comment:
Usage of `__typeof__` vs `typeof` could also be made a compile-time switch. However, since `embUnit` is also using `__typeof__` and we never had a problem with that, I don't see any problems coming our way.